From 0f0a483ad6e29bdb53f47dbb856161a2b5e36100 Mon Sep 17 00:00:00 2001
From: maniv-ami <manikandanv@ami.com>
Date: Mon, 5 May 2025 20:52:58 +0530
Subject: [PATCH 56/56] Ipmbbridged adds generator ID to payload that is from
 ipmb channel

Description:
Ipmb channel provide only up to 7 bytes without generator ID.
So the generator ID for ipmb channel is supposed to come from the ipmb requesters slave address.

---
 ipmbbridged.cpp | 12 ++++++++++++
 ipmbbridged.hpp |  6 ++++++
 2 files changed, 18 insertions(+)

diff --git a/ipmbbridged.cpp b/ipmbbridged.cpp
index 5e97ae4..c726c0c 100644
--- a/ipmbbridged.cpp
+++ b/ipmbbridged.cpp
@@ -437,6 +437,18 @@ void IpmbChannel::processI2cEvent()
         auto ipmbMessageReceived = IpmbRequest();
         ipmbMessageReceived.i2cToIpmbConstruct(ipmbFrame, r);
 
+	/* Ipmb channel provide only up to 7 bytes without generator ID so
+         * the generator ID for ipmb channel is supposed to come from the
+         * ipmb requesters slave address.
+         */
+        if (ipmbMessageReceived.netFn == netFnPlatformEvent &&
+            ipmbMessageReceived.cmd == cmdPlatformEvent)
+        {
+            // insert the generator ID into the head of the received packet data
+            ipmbMessageReceived.data.insert(ipmbMessageReceived.data.begin(),
+                                            ipmbFrame->Header.Req.rqSA);
+        }
+
         int devId = getDevIndex();
 
         std::map<std::string, std::variant<int>> options{
diff --git a/ipmbbridged.hpp b/ipmbbridged.hpp
index a0247c7..0d90f93 100644
--- a/ipmbbridged.hpp
+++ b/ipmbbridged.hpp
@@ -87,6 +87,12 @@ constexpr uint8_t ipmbNetFnResponseMask = 0x01;
 constexpr uint8_t ipmbLunMask = 0x03;
 constexpr uint8_t ipmbRsLun = 0x0;
 
+/**
+ * @brief Platform event command netFn and command
+ */
+constexpr uint8_t netFnPlatformEvent = 0x04;
+constexpr uint8_t cmdPlatformEvent = 0x02;
+
 /**
  * @brief Ipmb setters
  */
-- 
2.41.0

